client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionDialog/InteractionEventForm/useInteractionsForm.ts

```diff
@@ -24,89 +22,127 @@ const useInteractionsForm = (props: useInteractionFormIncome): useInteractionFor
-                        onDialogClose();
-                        alertError('לא ניתן היה ליצור אירוע חדש');
-                        setIsLoading(false);
-                    })
-                }
+            setIsLoading(true);
+            if (interactionsDataToSave[InteractionEventDialogFields.ID]) {
+                const updateInteractionsLogger = logger.setup('Update Interaction')
+                updateInteractionsLogger.info('launching update interaction request', Severity.LOW);
+                axios.post('/intersections/updateContactEvent', parsedData)
+                .then((response) => {
+                    if (response.data?.data?.updateContactEventFunction) {
+                        updateInteractionsLogger.info('updated interaction successfully', Severity.LOW);
+                        const eventId = response.data.data.updateContactEventFunction.integers[0]
+                        Promise.all([saveGroupedInvestigations(eventId), saveBankContacts(eventId)]).then(() => {
+                            loadInteractions();
+                            loadInvolvedContacts();
+                            onDialogClose();
+                            setIsLoading(false);
+                        }).catch(() => {
+                            onDialogClose();
+                            alertError('לא ניתן היה לעדכן אירוע');
+                            setIsLoading(false);
+                        });
+                    }
+                })
+                .catch((error) => {
+                    updateInteractionsLogger.error(`got error from server: ${error}`, Severity.HIGH);
+                    alertError('לא ניתן היה לשמור את השינויים');
+                    setIsLoading(false);
+                })
             } else {
-                alertError('ישנם תזים כפולים בטופס ולכן לא ניתן לשמור');
-                setIsLoading(false);
+                const createInteractionsLogger = logger.setup('Create Interaction');
+                createInteractionsLogger.info('launching create interaction request', Severity.LOW);
+                axios.post('/intersections/createContactEvent', parsedData)
+                .then((response) => {
+                    if (response.data?.data?.updateContactEventFunction) {
+                        createInteractionsLogger.info('created interaction successfully', Severity.LOW);
+                        const eventId = response.data.data.updateContactEventFunction.integers[0]
+                        Promise.all([saveGroupedInvestigations(eventId), saveBankContacts(eventId)]).then(() => {
+                            loadInteractions();
+                            loadInvolvedContacts();
+                            onDialogClose();
+                            setIsLoading(false);
+                        }).catch(() => {
+                            onDialogClose();
+                            alertError('לא ניתן היה ליצור אירוע חדש');
```

Comment:
save in const because of multiple use

Reply:
fixed [here](https://github.com/CoronaTeamMOD/CoronaInvestigation/pull/1360/commits/5d819ee29d3b0a7dbf24e384b786f593d2873e9c)